internal/cmd/shell/shell.go

```diff
@@ -182,7 +188,11 @@ second argument:
 				"-t", // the -s (silent) flag disables tabular output, re-enable it.
 				"-h", host,
 				"-P", port,
-				"-D", "@primary",
+			}
+			if replica {
+				mysqlArgs = append([]string{"--no-defaults"}, mysqlArgs...)
```

Comment:
Is there a reason you chose to _prepend_ `--no-defaults`?

Reply:
Because it didn't work otherwise 🤷🏾